shunting-yard.h

```diff
@@ -188,18 +188,43 @@ struct rpnBuilder {
   // * * * * * Static parsing helpers: * * * * * //
 
   // Check if a character is the first character of a variable:
-  static inline bool isvarchar(const char c) {
-    return isalpha(c) || c == '_';
+  // Returns the byte-length of the character
+  // (rest is needed for UTF8 characters)
+  static inline unsigned char isvarchar(const char c, const char** rest) {
+    unsigned char utf8charsize = isUTF8char(c, rest);
+    if(utf8charsize == 0) return isalpha(c) || c == '_' ? true : false;
+    return utf8charsize;
+  }
```

Comment:
For the purpose of validating utf8 chars, I think a better signature for this function would be like this:

```suggestion
  static inline uint8_t isvarchar(const char* str) {
    unsigned char utf8charsize = isUTF8char(str);
    if(utf8charsize == 0) return isalpha(*str) || *str == '_' ? true : false;
    return utf8charsize;
  }
```

This way we are returning a number instead a character, which is more semantic, and we are not passing a pointer to `char*` which implied we would make changes on this input.

Let me explain why and when I use `char**`. When parsing strings it is often necessary to know, after parsing some token, from where to continue. For doing that we have several options, e.g.:

1. We can return the number of characters parsed, and then skip these characters outside the function.
2. Or, since we are using C++ we can use the `char*` as a counter that always points to the next character to be parsed, and then when a subfunction is going to parse something we can conveniently pass this `char*` by reference so that the subfunction can do both things: parse the token and set the pointer to the next character. For this reason several of the functions in this code receive a `const char** rest`.

Now you might be confused: If I am talking about passing by reference, why not use `const char*& rest` instead? I don't remember where I read it, but there is a good practice regarding readability when we decide to use references or pointers in function arguments.

The is that when passing the address of a variables it is obvious to the person reading the function call that this variables might be changed by the function, i.e.:

```c++
v = parseVar(&str)
```

However, if I was passing this string argument by reference it would be implicit, e.g.:

```c++
v = parseVar(str)
```

Which hides information from the reader, which is not good.

So the criteria for writing functions that receives arguments by reference or by value is this:

- If you won't change the input variable, and just want to prevent an unnecessary copy of it, the argument can be a constant reference.
- If you are planning on modifying this input (thus making it an output parameter) then it is better to expect a pointer to type you are expecting, this will force the user of the function to use the `&` operator, which makes it obvious for the reader that that function call might modify this input parameter.

Reply:
However, I think we need to be sure that counting the number of characters is important, and that it must be in this function, because this is a public function that might already be used inside other projects. So although I am not super worried about backwards compatibility I would prefer to maintain it if possible.

For example, maybe we could instead of changing the behavior of this function which is quite intuitive, just call the isUTFchar function on the 2 places where we would like to change this behavior.